clients/fides-js/src/fides.ts

```diff
@@ -200,6 +200,7 @@ _Fides = {
     customOptionsPath: null,
     preventDismissal: false,
   },
+  shouldResurfaceConsent: false,
```

Comment:
hm why does this need to be on the Fides object?

Reply:
I'm not totally following the reasoning behind the `shouldResurfaceConsent` refactor?

Reply:
See https://github.com/ethyca/fides/pull/4527/files#diff-efe8c21098d15fb79a27d45475d1158e92c8ee59f04d2b2c6393bcf230f64019R351. We need a way to reliably determine if not all notices have previous consent (thus we know to resurface the banner) before we overwrite the cookie with any default vals. A good example would be for GPC.

To illustrate:

1. Cookie has consent vals "a" = true, "b" = false.
2. User starts a new session on the site, and when fides loads, the experience returns notices `a`, `b`, and a new notice, `c`, which is GPC enabled.
3. In the init of fides, we assign `previously_consented` to `true` for both `a` and `b`, but `c` is `false` since `c` is not found in the cookie. Now know we have to resurface the banner.
4. Further down in the init, we overwrite the cookie based on the experience. See-. When this happens, the cookie will now have "a"=true ,"b"=false, "c"=false (GPC applied). Now we can still use our `shouldResurfaceConsent` var to resurface consent for this user, even when GPC has been applied to the net new notice.

Reply:
what if we didn't overwrite the cookie with the default vals? I can imagine a flow like:

1. Page loads, we have a cookie and an experience with notices. The cookie might not have all the keys of the notice
2. `shouldResurfaceConsent` does its thing as it did before
3. in the `*Overlay` components, when generating the initial values, if the cookie doesn't have a value, then we grab the notice's default preference
4. only once we save, do we write the "default" preference to the cookie

Reply:
Moving from slack thread to GH for visibility:

Overall we agree that not overwriting the cookie would be simpler. Allison suggested:
* Override vals- we would still overwrite the cookie for this, and doesn't affect approach here
* GPC- shift the “setting” of the notice current preferences to render time as opposed to initializing the cookie with it
* net new consent keys from client-side experience- same approach as GPC, setting current preferences at render time


2 things I pointed out we need to consider:

1. We automatically save user prefs for GPC during init (before render time, see `automaticallyApplyGPCPreferences` fn). If we overwrite the cookie here we'd no longer be able to determine if we should resurface consent during render time. 

2. We also need to ensure this won't have consequences with actual consent compliance. When we dispatch our final FidesInitialized event at the end of init, this triggers consent to be pushed to our datalayer:
```
pushFidesVariableToGTM({
      type: "FidesInitialized",
      detail: {
        consent: window.Fides.consent,
        fides_meta: window.Fides.fides_meta,
        identity: window.Fides.identity,
        tcf_consent: window.Fides.tcf_consent,
      },
    });
```
And we actually use our cookie to set the window.Fides.consent obj, so we need to be sure this cookie obj reflects all the calculated / default consent vals we show on the FE too:
```
// return an object with the updated Fides values
  return {
    consent: cookie.consent,
    fides_meta: cookie.fides_meta,
    identity: cookie.identity,
    fides_string: cookie.fides_string,
    tcf_consent: cookie.tcf_consent,
    experience: effectiveExperience,
    shouldResurfaceConsent: resurfaceConsent,
    geolocation,
    options,
    initialized: true,
  };
```